File: java/lab5/src/commands/ExitCommand.java
package src.commands;

import src.baseabstractions.Command;
import src.basecollection.CollectionManager;
import src.baseobjects.Product;

/**
 * Command to exit the program without saving.
 *
 * @param <K> the type of keys used in the collection, must be Comparable
 * @param <V> the type of values stored in the collection, must extend Product
 */
public class ExitCommand<K extends Comparable<K>, V extends Product> extends Command<K, V> {

    /**
     * Constructs an ExitCommand with the specified collection manager.
     *
     * @param collectionManager the collection manager to operate on
     */
    public ExitCommand(CollectionManager<K, V> collectionManager) {
        super(collectionManager);
    }

    /**
     * Executes the exit command.
     * Terminates the program without saving the collection to file.
     *
     * @param args command arguments (expects exactly one argument - the command
     *             name)
     * @throws IllegalArgumentException if the number of arguments is incorrect
     */
    public void execute(String[] args) {
        if (args.length == 1) {
            getCollectionManager().exit();
        } else
            throw new IllegalArgumentException("Number of arguments is wrong.");
    }

    /**
     * Returns a description of the command.
     *
     * @return "exit : завершить программу (без сохранения в файл)" - command
     *         description in Russian
     */
    public String getDescription() {
        return "exit : завершить программу (без сохранения в файл)";
    }
}

File: java/lab6/client/src/main/kotlin/commands/ExitCommand.kt
package commands

import abstractions.Command
import kotlinx.serialization.Serializable
import kotlinx.serialization.Transient
import objects.Product
import kotlin.system.exitProcess

@Serializable
class ExitCommand<K : Comparable<K>, V : Product> : Command<K, V>() {
    @Transient
    override val description = "exit : завершить программу (без сохранения в файл)"

    @Transient
    override val isShouldBeSent = true

    override fun start(args: Array<String>) {
        if (args.size != 1) {
            throw IllegalArgumentException("Number of arguments is wrong.")
        }
    }

    override fun finish() {
        println("The job is done, goodbye!")
        exitProcess(0)
    }
}